tiledb/cf/core.py

```diff
@@ -384,107 +385,54 @@ def __exit__(self, exception_type, exception_value, exception_traceback):
-        return tiledb.open(array_uri, mode=mode, key=array_key, ctx=self.ctx)
-
-    def open_array(self, array_name: str, mode: str = "r") -> tiledb.Array:
-        """Open one array from the group in the specified mode.
-        """
-        array = self.get_open_uri_key(array_name, mode)
-        if array_name not in self.open_arrays:
-            self.open_arrays[array_name] = array
-        return array
-
-    def array_schema(self, array_name):
-        return self.group_schema.get(array_name)
-
-    def array_metadata(self, array_name: str) -> ArrayMetadata:
-        with self.get_open_uri_key(array_name) as array:
-            meta = ArrayMetadata(array.meta)
-        return meta
-
-    def attr_metadata(self, attr: str) -> Union[AttrMetadata, Dict[str, AttrMetadata]]:
-        """Attribute metadata wrapper for the attribute accessed at initialization.
-
-        Can only be used if an attribute or an array with exactly one attribute was
-        specified when initializing this group.
-        """
-        attr_metadata = dict()
-        array_names = self.group_schema.arrays_with_attr(attr)
-        if len(array_names) == 0:
-            return "no attr metadata found"
-        for name in array_names:
-            if name != METADATA_ARRAY_NAME:
-                with self.get_open_uri_key(name) as array:
-                    metadata = AttrMetadata(array.meta, attr)
-                    attr_metadata[name] = metadata
-        if len(array_names) == 1:
-            return attr_metadata[array_names[0]]
-        return attr_metadata
+        if self._metadata_array is not None:
+            self._metadata_array.close()
+        for array in self._opened_arrays:
+            array.close()
+        self._opened_arrays.clear()
 
     @property
     def has_metadata_array(self) -> bool:
         """Flag that is true if there a metadata array for storing group metadata."""
-        return self.metadata_array is not None
+        return self._metadata_array is not None
 
     @property
     def meta(self) -> Optional[tiledb.Metadata]:
         """Metadata object for the group, or ``None`` if no array to store group
         metadata exists."""
-        if self.metadata_array is None:
+        if self._metadata_array is None:
             return None
-        with self.metadata_array as meta_array:
-            meta = Metadata(meta_array.meta)
-        return meta
+        return self._metadata_array.meta
+
+    def open_array(
+        self, array: Optional[str] = None, attr: Optional[str] = None
+    ) -> tiledb.Array:
+        if array is None and attr is None:
+            raise ValueError(
+                "Cannot open array. Either an array or attribute must be specified."
+            )
+        if array is None:
+            array_names = self._group_schema.arrays_with_attr(attr)
+            if array_names is None:
+                raise KeyError(f"No attribute with name '{attr}' found.")
+            if len(array_names) > 1:
```

Comment:
In general, I this should work with a context manager. For example,
```
with group.open_array(...) as array:
    ...
```
I'm open to handling multiple arrays, if you can think of a way to use them together with the context manager. Maybe change it to a `NotImplementedError` instead of a `ValueError`?